### DISCUSSION TITLE
Minimap

### DISCUSSION THREAD

**Discussion by aaalizk** [boscop-fr/assistant-rgaa#34] [Ideas]
Il serait intéressant déjà d'indiquer le nombre d’occurrences pour éviter de devoir tout scroller sans raison. Mais en plus de les afficher sur le côté dans une minimap similaire a de nombreux IDE comme VScode ou Atom : 

![Capture d’écran 2022-03-03 162015](https://user-images.githubusercontent.com/69298848/156594621-54ecb385-fece-43f6-bf74-df141fb848d3.png)

Une fonction en plus serait de pouvoir se déplacer dans la page aux cliques sur cette minimap pour pouvoir cibler directement les endroits indiqués.

![vscode-minimap-full-view](https://user-images.githubusercontent.com/69298848/156595508-3ed5733b-627a-49b1-b236-b112ecbca704.gif)

**Comment by felixgirault** [comment]
L'idée me plaît bien !
Côté technique, je pense qu'il serait faisable d'indiquer les occurrences sur une scrollbar, comme le font également certains IDEs.

![scrollbar](https://user-images.githubusercontent.com/1544510/156599976-7a413623-0817-4671-a6bf-9165eac59682.png)

Par contre une minimap avec preview du contenu serait beaucoup plus compliquée (notamment en terme de performances), et je me demande si ça aurait une réelle plus-value.

**Comment by sbonaventure** [comment]
Une idée complémentaire serait d'avoir en bas de l'extension, un composant indiquant le nombre d'éléments mis en avant, et des boutons pour passer aux éléments suivant/précédents

**Comment by felixgirault** [comment]
Une fonctionnalité expérimentale disponible sur Firefox permet de générer des images depuis un élément HTML, ça semble particulièrement adapté aux minimaps :+1: 
https://developer.mozilla.org/en-US/docs/Web/CSS/element
https://uiw.tf/minimap

**Comment by felixgirault** [comment]
La version 2 de l'assistant propose une minimap.
